tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1143,6 +1168,66 @@ def test_login_failure_rhsso_user_if_internal_user_exist(
         assert error.typename == "NavigationTriesExceeded"
 
 
+@destructive
+def test_user_permissions_rhsso_user_multiple_group(
+    enable_external_auth_rhsso,
+    rhsso_setting_setup,
+    session,
+    module_org,
+    module_loc,
+    groups_teardown,
+    rhsso_groups_teardown,
+):
+    """Verify the permissions of the rhsso user, if it exists in multiple groups (admin/non-admin).
+        The rhsso user should contain the highest level of permissions from among the
+        multiple groups. In this case, it should contain the admin permissions.
+
+    :id: 782926c0-d109-41a0-af7a-bffd658f59d7
+
+    :CaseImportance: Medium
+
+    :steps:
+        1. create sat_users and sat_admins usergroups with non-admin and admin
+            permissions respectively
+        2. assigned these groups to rhsso user
+        3. verify the permission of the rhsso user in Satellite
+
+    :expectedresults: external rhsso user have highest level of permissions from among the
+        multiple groups.
+    """
+    username = settings.rhsso.rhsso_user
+    location_name = gen_string('alpha')
+    login_details = {
+        'username': username,
+        'password': settings.rhsso.password,
+    }
+    user_permissions = {'Katello::ActivationKey': PERMISSIONS['Katello::ActivationKey']}
+    katello_role = entities.Role().create()
+    create_role_permissions(katello_role, user_permissions)
+
+    group_names = ['sat_users', 'sat_admins']
+    arguments = [{'role': katello_role}, {'admin': 1}]
+    for group_name, argument in zip(group_names, arguments):
+        # adding/creating rhsso groups
+        create_group(group_name=group_name)
+        update_rhsso_user(username, group_name=group_name)
+        argument['name'] = group_name
+
+        # creating satellite external groups
+        user_group = make_usergroup(argument)
+        make_usergroup_external(
+            {'auth-source-id': '3', 'user-group-id': user_group['id'], 'name': group_name}
```

Comment:
Is it always going to be `'3'`? Could we obtain this value dynamically?
Otherwise, that might be good candidate for constant.

Reply:
Yes I think we can get it, will update based on that